chainercv/links/model/sequential_feature_extractor.py

```diff
@@ -33,42 +31,47 @@ class SequentialFeatureExtractor(chainer.Chain):
         >>> import collections
         >>> import chainer.functions as F
         >>> import chainer.links as L
-        >>> layers = collections.OrderedDict([
-        >>>     ('l1', L.Linear(None, 1000)),
-        >>>     ('l1_relu', F.relu),
-        >>>     ('l2', L.Linear(None, 1000)),
-        >>>     ('l2_relu', F.relu),
-        >>>     ('l3', L.Linear(None, 10))])
-        >>> model = SequentialFeatureExtractor(layers, ['l2_relu', 'l1_relu'])
+        >>> model = SequentialFeatureExtractor()
+        >>> with model.init_scope():
+        >>>     model.l1 = L.Linear(None, 1000)
+        >>>     model.l1_relu = F.relu
+        >>>     model.l2 = L.Linear(None, 1000)
+        >>>     model.l2_relu = F.relu
+        >>>     model.l3 = L.Linear(None, 10)
+        >>> model.layer_names = ('l2_relu', 'l1_relu')
         >>> # These are outputs of layer l2_relu and l1_relu.
         >>> feat1, feat2 = model(x)
         >>> # The layer_names can be dynamically changed.
         >>> model.layer_names = 'l3'
         >>> # This is an output of layer l1.
         >>> feat3 = model(x)
 
-    Args:
-        layers (collections.OrderedDict of callables):
-            Callable objects called in the forward pass.
+    Params:
         layer_names (string or iterable of strings):
             Names of layers whose outputs will be collected in
             the forward pass.
 
     """
 
-    def __init__(self, layers, layer_names=None):
+    def __init__(self):
         super(SequentialFeatureExtractor, self).__init__()
-
-        if not isinstance(layers, collections.OrderedDict):
-            raise ValueError('layers need to be collections.OrderedDict')
-        self._layers = layers
-
-        self.layer_names = layer_names
-
-        with self.init_scope():
-            for name, layer in self._layers.items():
-                if isinstance(layer, chainer.Link):
-                    setattr(self, name, layer)
+        self._order = list()
```

Comment:
Why do you prefer `_layers` over `_order`?
`_order` sounds like a list of integers. It doe not imply that it contains links and functions.